### DISCUSSION TITLE
App's package dependencies in node_modules are empty - Dockerfile

### DISCUSSION THREAD

**Discussion by AlexandrosMelissas** [vercel/turborepo#4173] [Help] [answered]
Hello, i have followed the with-docker example to the letter. The only problem I've had so far is in these lines:

```
COPY --from=builder /app/out/json/ .
COPY --from=builder /app/out/yarn.lock ./yarn.lock
RUN yarn install
```

Here we copy json folder of pruned version of turborepo which contains only the package.json files of each app and it's package/lib dependencies (e.x. tsconfig) which include only package.json files as well. 

The problem is every time i install and then exec the container with bash, i can see that /app/apps/my-app/node_modules/my-package has just the package.json. It doesn't have the built files because the package is empty in the first place. I am very confused as to why a yarn install with empty dependencies would work anyway. Then the example does this:
```
# Build the project
COPY --from=builder /app/out/full/ .
RUN yarn turbo run build --filter=web...
``` 

Now we have something like this:
|──  app
|────   apps
|──────  my-app
|───────  node_modules
|────────   my-lib@1.0.0
|─────────  package.json (only this file)
|──── packages
|───── my-package
|────── ...files (now we have them cause we copied full folder)

So my question is how exactly is this supposed to work? Am i missing something?

**Comment by AlexandrosMelissas** [comment, marked as answer]
I think i managed to fix it. See [issue](https://github.com/vercel/turbo/issues/4179)

**Reply by Fish1** [reply]
Oh thank you. I've been trying to work around this in my CI.
